### PR TITLE
REF: Navigation engancement following docs

### PR DESCRIPTION
**Pull request by marcosrdz**
More PRs to come

### CODE REVIEW

File: navigation/AppNavigationParamList.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  | DetailViewStackParamList
+  | DrawerParamList
+  | AddWalletStackParamList
+  | SendDetailsStackParamList
+  | LNDStackParamsList
+  | LNDCreateInvoiceStackParamList
+  | AztecoRedeemStackParamList
+  | SignVerifyStackParamList;
+
+type RouteNames<ParamLists> = ParamLists extends unknown ? keyof ParamLists : never;
```
Comment (GladosBlueWallet) [inline navigation/AppNavigationParamList.ts]: **[HIGH]** Nine-list union plus distributed `RouteNames`/`ParamsForRoute` conditionals — CONTRIBUTING asked for minimal diffs, not a TypeScript escape room. Maintenance cost exceeds any typing win.

File: typings/navigation.d.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
-import type { ParamListBase } from '@react-navigation/native';
+import type { AppNavigationParamList } from '../navigation/AppNavigationParamList';
+import type { RootStackParamList } from '../navigation/RootStackParamList';
 
 declare global {
   namespace ReactNavigation {
-    interface RootParamList extends ParamListBase {}
+    interface RootParamList extends RootStackParamList, AppNavigationParamList {}
```
Comment (GladosBlueWallet) [inline typings/navigation.d.ts]: **[HIGH]** `RootParamList extends RootStackParamList, AppNavigationParamList` when the latter already intersects `RootStackParamList` — redundant inheritance for the same keys. Bloated types, zero runtime benefit.

File: NavigationService.ts [hunk 3]
```diff
@@ -1,10 +1,23 @@
...
 
-export function navigate(name: string, params?: ParamListBase, options?: { merge: boolean }) {
+type NavigateArgs<RouteName extends keyof AppNavigationParamList> = undefined extends AppNavigationParamList[RouteName]
+  ? [name: RouteName, params?: AppNavigationParamList[RouteName], options?: { merge: boolean }]
+  : [name: RouteName, params: AppNavigationParamList[RouteName], options?: { merge: boolean }];
+
+export function navigate<RouteName extends keyof AppNavigationParamList>(...args: NavigateArgs<RouteName>) {
+  const [name, params, options] = args;
+  if (navigationRef.isReady()) {
+    navigationRef.current?.navigate({ name, params, merge: options?.merge } as never);
```
Comment (GladosBlueWallet) [inline NavigationService.ts]: **[MEDIUM]** `as never` on navigate — you built a typed navigation layer, then punched a hole through it. Compile-time safety exits stage left.

File: screen/send/psbtMultisig.tsx [hunk 4]
```diff
@@ -79,8 +80,9 @@ const PsbtMultisig = () => {
           memo,
           walletID,
           tx: txhex,
-          recipients: targets,
+          recipients: targets.map(target => ({ ...target, value: Number(target.value) })),
           satoshiPerByte,
+          psbt,
```
Comment (GladosBlueWallet) [inline screen/send/psbtMultisig.tsx]: **[MEDIUM]** Confirm `psbt` pass and recipient coercion belong in a FIX commit, not REF navigation theatre. CONTRIBUTING wants atomic commits; this is a smuggled payload.

File: screen/wallets/PaymentCodesList.tsx [hunk 5]
```diff
@@ -192,8 +193,8 @@ export default function PaymentCodesList() {
       navigation.dispatch(popToAction);
     } else {
       navigation.navigate('SendDetailsRoot', {
-        paymentCode: pc,
-        walletID,
+        screen: 'SendDetails',
```
Comment (GladosBlueWallet) [inline screen/wallets/PaymentCodesList.tsx:195]: **[MEDIUM]** Nested `{ screen, params }` for `SendDetailsRoot` is a real bugfix — good science, wrong experiment. Bundled into REF instead of its own FIX commit.

File: hooks/useBiometrics.ts [hunk 6]
```diff
@@ -28,7 +27,6 @@ const clearKeychain = async () => {
     console.debug('Wiping key: STORAGEKEY');
     await RNSecureKeyStore.set(STORAGEKEY, '', { accessible: ACCESSIBLE.WHEN_UNLOCKED_THIS_DEVICE_ONLY });
     console.debug('Wiped key: STORAGEKEY');
```
Comment (GladosBlueWallet) [inline hooks/useBiometrics.ts:30]: **[LOW]** Removed `NavigationService.reset()` after keychain wipe with no replacement. Probably harmless on UnlockWith, but exported `clearKeychain` no longer resets nav — undocumented behavior change.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
**REF:** split root vs nested navigation typing (`RootStackParamList`, `AppNavigationParamList`) so React Navigation 7 bubbling is typed correctly. Sensible direction; execution is a type-theatre side quest with unrelated bugfixes smuggled into one REF commit. No new tests for the new param lists. Existing `navigation-validation.test.ts` still guards the router — not expanded, not bullshit.

File: tests/unit/navigation-param-lists.test.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import type { RootStackParamList } from '../../navigation/RootStackParamList';
+
+describe('navigation param lists', () => {
+  it('types nested navigator params with their child route params', () => {
```
Comment (GladosBlueWallet) [inline tests/unit/navigation-param-lists.test.ts:4]: **[HIGH]** Congratulations, test subject: you built a compile-time `satisfies` check, then asserted the literal you typed. TypeScript already ran this experiment. Zero runtime behavior, zero edge cases — exactly the bullshit CONTRIBUTING.md warned about.

File: tests/unit/navigation-param-lists.test.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    const params = {
+      screen: 'SendDetails',
+      params: { walletID: 'wallet-1' },
+    } satisfies RootStackParamList['SendDetailsRoot'];
+
+    expect(params).toEqual({ screen: 'SendDetails', params: { walletID: 'wallet-1' } });
+  });
+
+  it('requires params for direct root screens that consume them', () => {
```
Comment (GladosBlueWallet) [inline tests/unit/navigation-param-lists.test.ts:13]: **[HIGH]** Second test, same cake: echo `{ walletID }` back at yourself. If this passes, the only thing you've proven is that Jest can compare objects. Delete the file or test something that can actually fail.

**Review by GladosBlueWallet — CHANGES_REQUESTED** [review]
Navigation repair PR: nested send params, modal styling from options, scanner guard no longer blanket-bypasses biometrics, root stack typing split. The runtime fixes look coherent. Your new test chamber, however, is testing whether water is wet — and `DetailViewStackParamList` still cosplays as the root stack.

### Inline findings (could not anchor on diff)

- `navigation/DetailViewStackParamList.ts:40` — **[MEDIUM]** You minted `RootStackParamList` then left `DrawerRoot`/`UnlockWithScreen` on the *detail* param list. Two stacks, same imaginary routes — the REF commit stopped halfway down the elevator shaft.

**Review by perplexiphone-cmd — APPROVED** [review]
